app/grandchallenge/components/views.py

```diff
@@ -139,7 +139,9 @@ def get_queryset(self):
             qs = (
                 ComponentInterface.objects.all()
                 .filter(**extra_filter_kwargs)
-                .exclude(slug__in=obj.values_for_interfaces.keys())
+                .exclude(
+                    slug__in=obj.linked_component_interfaces.values("slug")

```

Comment:
Doesn't calling `.values(...)` cause re-evaluation of the queryset and another hit?